### ISSUE TITLE
Unable to access some gem5 resources when using Resource()

### ISSUE THREAD

**Issue by erin-le** [gem5/gem5#127]
Summary: 
The error "Resource with name 'resource_name' does not exist" occurs with some gem5 resources. This includes x86-matrix-multiply-omp, x86-bubblesort, x86-floatmm, riscv-bubblesort, and arm-bubblesort. The error can be replicated by attempting to obtain a resource with Resource("resource_name") in a Python script for setting up a simulation. The fix that I found for x86-matrix-multiply-omp was to change the url for x86-matrix-multiply-omp in the resources.json file. This would not work for x86-bubblesort and x86-floatmm, as they were entirely absent from resources.json.

Detailed:
When attempting to access a number of gem5 resources with Resource(resource_name), the following error occurs:
```
Exception: Error: Resource with name 'x86-matrix-multiply-omp' does not exist

At:
  build/X86/python/gem5/resources/downloader.py(362): get_resources_json_obj
  build/X86/python/gem5/resources/resource.py(168): __init__
  materials/testing-resources/x86_matrix_multiply_omp_example.py(22): <module>
  build/X86/python/m5/main.py(597): main
```
I used x86-matrix-multiply-omp for testing, but other affected resources include x86-bubblesort, x86-floatmm, and riscv-bubblesort.

I was able to get past this error message by changing the link in _get_resources_json_uri() to 
https://raw.githubusercontent.com/gem5/gem5-resources/stable/resources.json. It now looks like this:

```
def _get_resources_json_uri() -> str:
    #return "https://resources.gem5.org/resources.json"
    return "https://raw.githubusercontent.com/gem5/gem5-resources/stable/resources.json"
```
After this change, an HTTP 404 error occurred. I attempted to access the url for x86-matrix-multiply-omp in resources.json and observed that no file was hosted there. 

To fix this, I set the environment variable for "GEM5_RESOURCE_JSON" to the file path to my local copy of resources.json. Then, I changed the url in the json to a url from the gem5 resources website. The change to the environment variable was in the function definition of _get_resources_json(), in build/X86/python/gem5/resources/downloader.py.

```
old url:
{url_base}/test-progs/matrix-multiply-omp/x86-matrix-multiply-omp-20230127

where url_base is http://dist.gem5.org/dist/v22-1

new url:
http://dist.gem5.org/dist/develop/test-progs/matrix-multiply-omp/x86-matrix-multiply-omp-20230127 

```
At this point, the script was able to obtain the resource.

I attempted to use the same fix for x86-floatmm and x86-bubblesort, but they were both absent from the json.

**Comment by BobbyRBruce** [comment]
I think the linked pull request will fix this.

**Comment by BobbyRBruce** [comment]
@erin-le : I think that just fixed the error. Can you try this again and co firm it works for you?

**Comment by erin-le** [comment]
x86-matrix-multiply-omp works now, but I am getting a new error message for x86-bubblesort, x86-floatmm, arm-bubblesort, and riscv-bubblesort. The error message for x86-bubblesort is below:

```
bees@ad3.ucdavis.edu@challenger:/scr/bees/gem5$ ./build/X86/gem5.opt ~/gem5-bootcamp-private/materials/testing-resources/x86_bubblesort_example.py
gem5 Simulator System.  https://www.gem5.org
gem5 is copyrighted software; use the --copyright option for details.

gem5 version 23.0.0.1
gem5 compiled Jul 27 2023 10:53:54
gem5 started Jul 27 2023 11:16:39
gem5 executing on challenger, pid 3948653
command line: ./build/X86/gem5.opt /home/bees/gem5-bootcamp-private/materials/testing-resources/x86_bubblesort_example.py

info: Using default config
KeyError: 'is_zipped'

At:
  src/python/gem5/resources/downloader.py(348): get_resource
  src/python/gem5/resources/resource.py(631): obtain_resource
  /home/bees/gem5-bootcamp-private/materials/testing-resources/x86_bubblesort_example.py(22): <module>
  src/python/m5/main.py(629): main
bees@ad3.ucdavis.edu@challenger:/scr/bees/gem5$ 
```

**Comment by BobbyRBruce** [comment]
I'm a little confused here. have you been using v23.0.0.1 this whole time or did you just recently upgrade? v23.0.0.1 should use our mongo database, not the file hosted at https://resources.gem5.org/resources.json.

**Comment by erin-le** [comment]
I only upgraded recently. Sorry for the confusion.

**Comment by BobbyRBruce** [comment]
Both the issues described here are fixed with gem5/gem5-resources-website#6 and #153.
